mteb/tasks/Retrieval/GermanDPRRetrieval.py

```diff
@@ -23,11 +24,17 @@ def description(self):
             "revision": "5129d02422a66be600ac89cd3e8531b4f97d347d",
         }
 
-    def _format_documents(self, docs, id_prefix=""):
+    @staticmethod
+    def _format_documents(docs, id_prefix="", existing_docs=None):
+        if existing_docs is None:
+            existing_docs = dict()
         result = {}
-        for i, (title, content) in enumerate(zip(docs["title"], docs["text"])):
-            id_value = f"{id_prefix}{i}"
-            formatted_content = content.split("===")[-1].replace("\n", " ").lstrip()
+        for i, (title, content) in enumerate(zip(docs['title'], docs['text'])):
+            formatted_content = content.split('==\n')[-1].replace('\n', ' ').lstrip()
+            if formatted_content in existing_docs:
+                id_value = existing_docs[formatted_content]
+            else:
+                id_value = f"{id_prefix}{i}"
```

Comment:
Don't we need something like this? Otherwise `existing_docs` always remains empty, I think.
```suggestion
                id_value = f"{id_prefix}{i}"
                existing_docs[formatted_content] = id_value
```